### PR TITLE
The Proxy.close() function doesn't close http connections immediately (close #89)

### LINKED ISSUE #89: The Proxy.close() function doesn't close http connections immediately
As a result after the following code
```js
var proxy1 = new Proxy(hostname, port1, port2);
proxy1.close();
var proxy2 = new Proxy(hostname, port1, port2);
```
the proxy1 connections are still available but proxy2 connections are not.

The cause of the problem in the `http` module (see [here](http://stackoverflow.com/questions/18874689/force-close-all-connections-in-a-node-js-http-server) and [here](http://stackoverflow.com/questions/14626636/how-do-i-shutdown-a-node-js-https-server-immediately)).

Simple examples to reproduce:

This code always returns '1' to the `http://localhost:1335` page:
```js
var http = require('http');

var server = http.createServer(function (req, res) {
    res.end('1');
});

server.listen(1335);
console.log('server started');
setTimeout(function () {
    server.close();

    server = http.createServer(function (req, res) {
        res.end('2');
    });

    server.listen(1335);
    console.log('server changed');
}, 5000);
```

This code returns '1' for the first 5 second, then returns '2':
```js
var http = require('http');

var server = http.createServer(function (req, res) {
    res.end('1');
});

server.listen(1335);

// Maintain a hash of all connected sockets
var sockets = {}, nextSocketId = 0;
server.on('connection', function (socket) {
    // Add a newly connected socket
    var socketId      = nextSocketId++;
    sockets[socketId] = socket;
    console.log('socket', socketId, 'opened');

    // Remove the socket when it closes
    socket.on('close', function () {
        console.log('socket', socketId, 'closed');
        delete sockets[socketId];
    });

    // Extend socket lifetime for demo purposes
    socket.setTimeout(4000);
});

setTimeout(function () {
    // Close the server
    server.close(function () {
        console.log('Server closed!');
    });
    // Destroy all open sockets
    for (var socketId in sockets) {
        console.log('socket', socketId, 'destroyed');
        sockets[socketId].destroy();
    }

    server = http.createServer(function (req, res) {
        res.end('2');
    });

    server.listen(1335);
}, 5000);
```

### PR DESCRIPTION
**Pull request by churkin**
\cc @AlexanderMoskovkin @miherlosev

### CODE REVIEW

**Comment by miherlosev** [comment]
lgtm

**Comment by churkin** [comment]
tests have passed

**Comment by AlexanderMoskovkin** [comment]
lgtm

File: src/proxy.js [hunk 1]
```diff
@@ -47,9 +45,27 @@ export default class Proxy extends Router {
         this.server1.listen(port1);
         this.server2.listen(port2);
 
+        this.sockets = [];
+
+        // BUG: https://github.com/superroma/testcafe-hammerhead/issues/89
```
Comment (inikulin) [inline src/proxy.js]: I think we need a `NOTE` with explanation here.

File: src/proxy.js [hunk 2]
```diff
@@ -47,9 +45,27 @@ export default class Proxy extends Router {
...
+        this._startSocketsCollecting();
         this._registerServiceRoutes();
     }
 
+    _closeSockets () {
+        this.sockets.forEach(socket => socket.destroy());
+    }
+
+    _startSocketsCollecting () {
```
Comment (inikulin) [inline src/proxy.js]: Let's rename it to `_startCollectSockets`

**Comment by inikulin** [comment]
\cc @churkin There are some minor style nits. Can you issue a separate pull request with fix as the fix?

**Pull request merged**